cdap-app-templates/cdap-etl/hydrator-spark-core/src/main/java/io/cdap/cdap/etl/spark/batch/RDDCollection.java

```diff
@@ -123,6 +126,54 @@ public RDDCollection(JavaSparkExecutionContext sec, JavaSparkContext jsc, SQLCon
       }
       seenRequired = seenRequired || toJoin.isRequired();
 
+      // UUID for salt column name to avoid name collisions
+      String saltColumn = UUID.randomUUID().toString();
+      if (joinRequest.isDistributionEnabled() && JoinDistribution.getSupportedJoinTypes().contains(joinType)) {
+
+        final String saltUDF = "salt";
+        final String prepareExplodeUDF = "prepare_explode";
+
+        //Array of [0,distributionFactor) to be used in the PrepareExplode step
+        Integer[] numbers =
+          IntStream.range(0, joinRequest.getDistribution().getDistributionFactor()).boxed().toArray(Integer[]::new);
+
+        // Register UserDefinedFunctions for use later
+        sqlContext.udf().register(prepareExplodeUDF, prepareExplodeUDF(numbers),
```

Comment:
is the name supposed to be unique? Seems like there would be issues if multiple salted joins are in the same pipeline, but they both use different distribution factors.

Reply:
could this be done without a UDF? If there's a way to give a constant value followed by an explode?

Reply:
Yes that is possible, I chose a UDF because that will probably be needed if we want to implement the option to filter which rows get exploded. But like we discussed offline, UDFs have some unknowns so I've updated this to use a constant value.